Docs: Add more information around setting up Discourse (#18)
vikaspotluri123/discourse-on-ghost: docs/index.md

Ghost's default tier names can get pretty long, and this can cause Discourse Groups to fail being created. You might want to increase the `max username length` setting to 30 since username and group lengths are both controlled by this. Here's a [Discourse discussion for reference](https://meta.discourse.org/t/group-name-character-limitation/31229)

#### Add private categories

DoG syncs your member's tiers with a corresponding Discourse group, which means you can use Discourse's [Private Categories](https://meta.discourse.org/t/how-to-create-private-categories-using-category-permission-security-settings/87678) feature to create gated sections of the forum for commenting or conversation.

#### A note on using Discourse for comments on Ghost

It's possible to [use Discourse for comments on your Ghost publication](https://ghost.org/integrations/discourse/).
However, Discourse's embed feature doesn't allow dynamically setting the category.
This can be an issue if e.g. you're looking to create gated content sections in your forum.

To work around this limitation:

 - After publishing your post, head over to the comments section on the live post

>  - Click `Start Discussion` to open the auto-generated thread

Author: ```suggestion
 - If it hasn't already, wait for `Loading Discussion...` to switch over to `Start Discussion` (takes less than a minute), then click on the latter to open the auto-generated thread
```

Reviewer: ` - Click `Start Discussion` to open the auto-generated thread. It might take a minute to load`

Ghost's default tier names can get pretty long, and this can cause Discourse Groups to fail being created. You might want to increase the `max username length` setting to 30 since username and group lengths are both controlled by this. Here's a [Discourse discussion for reference](https://meta.discourse.org/t/group-name-character-limitation/31229)

#### Add private categories

> Private categories allow you to limit topics posted in the category to a specific group. Since DoG syncs member's tiers to the corresponding Discourse group, you can use private categories to create gated content and commenting. Refer to [this Discourse Meta topic](https://meta.discourse.org/t/how-to-create-private-categories-using-category-permission-security-settings/87678) for more information about creating and securing private categories.

Author: ```suggestion
Private categories allow you to limit topics posted in a particular category to a specific group. Since DoG syncs member's tiers to the corresponding Discourse group, you can use private categories to create gated content and commenting. Refer to [this Discourse Meta topic](https://meta.discourse.org/t/how-to-create-private-categories-using-category-permission-security-settings/87678) for more information about creating and securing private categories.
```

Reviewer: ```
DoG sync your member's tiers with a corresponding Discourse group, which means you can use Discourse's [Private Categories feature] to create gated sections of the forum for commenting or conversation.
```

Author: Yup, although "sync" should be "syncs"?

Reviewer: Thanks, fixed!

> #### Using Discourse for comments on Ghost

Author: ```suggestion
#### Maintaining your private categories setup
```

Reviewer: I'm still not convinced - there don't seem to be that many publications with multiple tiers that would make this heading applicable. A majority of the installation instructions are "generally relevant", and I'd like to do the same here if possible

Author: So is the problem that categories is in plural? Could it be solved by using "category/categories setup", supposing that wouldn't be too long of a title?

My main issue with what is currently there is that it's too broad, and kinda sounds as if it's supposed to be how to Discourse for, well, commenting on Ghost, as opposed to "Using DoG" or something.

Reviewer: You're right, but in this case I think it's useful to link to the Ghost integration. Maybe add the "a note on using..." Qualifier like I mebntioned yesterday?

Author: Definitely link to it, nothing wrong with that. I think I may have even originally added it in my original changes? Or maybe not.

> It's possible to [use Discourse for commenting on your Ghost publication](https://ghost.org/integrations/discourse/), but automated topic creation can be limited in the following scenario:

Author: ```suggestion
While it's possible to [use Discourse for commenting on your Ghost publication](https://ghost.org/integrations/discourse/), automated topic creation can be limited due to a limitation in Discourse embeds — Discourse is currently only able to deposit topics into one pre-defined category. This can be an issue due to the need (for tiers and some membership-utilising sites) for topics to be placed into specific categories (each category associated with a particular Discourse group, tied to a Ghost tier). Here's a publishing flow to work around this limitation:
```

Author: The four bullet points were somewhat confusing, and I think this and the follwing two sentences can be combined into this initial one. Perhaps the last part (about "specific categories" and "a particular Discourse group, tied to a Ghost tier") can be amended a bit.

> It's possible to [use Discourse for commenting on your Ghost publication](https://ghost.org/integrations/discourse/), but automated topic creation can be limited in the following scenario:

Author: ```suggestion
While it's possible to [use Discourse for commenting on your Ghost publication](https://ghost.org/integrations/discourse/), automated topic creation can be limited if you need to use more than one category/group (for setups with multiple tiers, setups with public posts _and_ members-only posts, etc.). This is due to a limitation in Discourse embeds in which Discourse is currently only able to deposit topics into one pre-defined category. This can be an issue due to the need (for the aforementioned setups) for topics to be placed into specific categories. Here's the publishing flow to work around this limitation:
```

Author: I think this sentence, the four bullet points, and the following two sentences can be consolidated into one, perhaps making it all clearer to understand.

Reviewer: ```
It's possible to [use Discourse for commenting on your Ghost publication](https://ghost.org/integrations/discourse/). However, Discourse's embed feature doesn't allow dynamically setting the topic. This can be an issue if e.g. you're looking to created gated content sections in your forum.

{workaround 1}

Since Discourse allows you to specify the topic based on the URL, you can also use Ghost's [Dynamic Routing](xxx) to create a "subdirectory" for each tier.
```

Author: Yup, although "created" should be "create".

Reviewer: Fixed. Once we figure out the other changes I'll commit them all at once

>  - You have tiers in Ghost

Author: ```suggestion
```

>  - The conversation topic for each post should be associated with a private Discourse category

Author: ```suggestion
```

>  - The Ghost tiers are used to gate access to the private Discourse category

Author: ```suggestion
```

>  - In other words (tl;dr): The topic associated to a comment thread isn't the same for every post

Author: ```suggestion
```

> This is due to a limitation in Discourse embeds - the topic is configured in the Discourse admin, not on the fly.

Author: ```suggestion
```

> Here's a publishing flow to work around this limitation:

Author: ```suggestion
```

>  - You have tiers in Ghost
>  - The conversation topic for each post should be associated with a private Discourse category
>  - The Ghost tiers are used to gate access to the private Discourse category
>  - In other words (tl;dr): The topic associated to a comment thread isn't the same for every post
> 
> This is due to a limitation in Discourse embeds - the topic is configured in the Discourse admin, not on the fly.
> 
> Here's a publishing flow to work around this limitation:

Author: ```suggestion
```

Author: I found this could be a bit confusing so condensed it into the couple of sentences above.

Reviewer: I see your point - maybe just [keep the tl;dr](https://github.com/vikaspotluri123/discourse-on-ghost/pull/18?notification_referrer_id=NT_kwDOAELjTLI1NDYwMjAxMDA0OjQzODM1NjQ#discussion_r1096492328)?

Author: TBH I'm not ever sure what's being said there. Aren't topics and comment threads one and the same? And isn't it stating the obvious that each post has its own topic / comment thread?

Reviewer: Updated, I accidentally put `topic` instead of `category` 😴

Author: Ah, okay. Perhaps I'd say switch it all around then.

> In other words (tl;dr): Posts aren't necessarily all associated with the same category.

Reviewer: Ah what I meant was just use the copy in the other thread, which is basically just the tl;dr

Author: Ah, you mean move things around, and place "tl;dr" somewhere? If so, sure?

Reviewer: Let me commit the changes and you can review again if there's an issue 😅

Author: Good idea!

 - Publish your post

>  - View your post, and scroll to the comments section

Author: ```suggestion
 - View your post, and scroll to the automatically generated comments section
```

>  - Create the Discourse topic for the post

Author: ```suggestion
```

Author: Discourse topics are automatically generated (as I added in the previous bullet point), so I figure this point should be deleted

>  - Click `Continue Discussion` to open the non-embedded version of the thread

Author: ```suggestion
 - Click `Start Discussion` to open the non-embedded version of the thread
```

Author: Before the first comment has been left it says "Start Discussion"

>  - Update the topic's category to the private category associated with the tier

Author: ```suggestion
 - Update the topic's category to the private category associated with members-only posts or with the associated tier
```

>  - View your post, and scroll to the comments section
>  - Create the Discourse topic for the post
>  - Click `Continue Discussion` to open the non-embedded version of the thread
>  - Update the topic's category to the private category associated with the tier

Author: ```suggestion
 - View your post, and scroll to the automatically generated comments section
 - Click `Start Discussion` to open the non-embedded version of the thread
 - Update the topic's category to the private category associated with members-only posts or with the assocaited tier
```

Author: It says to "Create the Discourse topic", but as it's automatically generated that's an incorrect instruction. To clarify (if needed) I added the words "automatically generated" in the bullet point above.

Before a comment has been left it says "Start Discussion", not "Continue Discussion".

In the last bullet point I amended I added in "members-only posts" because the user might have a site with public posts and members-only posts, and have no tiers.

>  - View your post, and scroll to the comments section
>  - Create the Discourse topic for the post
>  - Click `Continue Discussion` to open the non-embedded version of the thread
>  - Update the topic's category to the private category associated with the tier

Author: ```suggestion
 - View your post, and scroll to the automatically generated comments section
 - Click `Start Discussion` to open the non-embedded version of the thread
 - Update the topic's category to the private category associated with members-only posts or with the associated tier
```

Author: It says to "Create the Discourse topic", but as it's automatically generated that's an incorrect instruction. To clarify (if needed) I added the words "automatically generated" in the bullet point above.

Before a comment has been left it says "Start Discussion", not "Continue Discussion".

In the last bullet point I amended I added in "members-only posts" because the user might have a site with public posts and members-only posts, and have no tiers.

Reviewer: > Before a comment has been left it says "Start Discussion", not "Continue Discussion".

This is the part I'm referring to - but it turns out that step 2 and 3 should be merged together 🙃

```
 - View your post, and scroll to the comments section
 - Click `Continue Discussion` to open the auto-generated thread
 - Set the topic's category to the private category associated with the tier
 ```

Author: So you're saying okay to what I suggested, with no amendments?

Reviewer: The copy I commented is my proposal after applying the fix I mentioned

Author: Ah, okay. But it needs to be 'Start Discussion', not 'Continue Discussion'. I didn't look too closely and got thrown off by thinking it was your original copy.

#### Add private categories

DoG syncs your member's tiers with a corresponding Discourse group, which means you can use Discourse's [Private Categories](https://meta.discourse.org/t/how-to-create-private-categories-using-category-permission-security-settings/87678) feature to create gated sections of the forum for commenting or conversation.

#### A note on using Discourse for comments on Ghost

It's possible to [use Discourse for comments on your Ghost publication](https://ghost.org/integrations/discourse/).

> However, Discourse's embed feature doesn't allow dynamically setting the category.

Author: ```suggestion
However, Discourse's embed feature doesn't allow for dynamically setting the category.
```

Reviewer: I think both are correct?

Author: I suppose. Flip a coin.

This can be an issue if e.g. you're looking to create gated content sections in your forum.

To work around this limitation:

 - After publishing your post, head over to the comments section on the live post

>  - Click `Continue Discussion` to open the auto-generated thread

Author: ```suggestion
 - Click `Start Discussion` to open the auto-generated thread
```

 - Set the topic's category to the private category associated with the tier

> Alternatively, since Discourse allows you to specify the category based on the URL, you can also use Ghost's [Dynamic Routing](https://ghost.org/docs/themes/routing/) feature to create a "subdirectory" for each tier.

Author: Are you suggesting that when adding multiple hosts to the Customize --> Embedding section of Discourse that each can be of the same domain but with a unique subdirectory appended to the URL, which can then automatically deposit posts in the same subdirectory into their own category?

Reviewer: I'm definitely suggesting it, but I think it should be tried before adding this... I misremembered what the Discourse embedding settings looks like 😕

Author: Confirmed. This works. (Not that I'd recommend actually utilising it unless one specifically wanted particular posts appearing in their own subdirectories.)
